### ISSUE TITLE
[iOS] App crash on Apple Pay confirmation

### ISSUE THREAD

**Issue by joshuaruesweg** [flutter-stripe/flutter_stripe#1180]
**Describe the bug**
The iOS app crashes, after the payment is confirmed within the Apple Pay dialog. 

The following error message is thrown: 

```stripe_ios/ApplePayButtonView.swift:123: Fatal error: Unexpectedly found nil while unwrapping an Optional value```

which refers to the following code: 

```swift
    func onOrderTracking(_ arguments: Dictionary<AnyHashable, Any>?) {
        channel.invokeMethod("onOrderTracking", arguments: arguments!)
    }
```

**To Reproduce**
1. Confirm an Apple Pay. 

**Smartphone / tablet**
 - Device: iPhone 14 Pro Max (Simulator on latest macOS)
 - OS: iOS 16.2
 - Package version: 9.0.0
 - Flutter version: 3.7.7

**Additional context**
A minimal working example: 

```dart
PlatformPayButton(
  onShippingMethodSelected: (method) {
    Stripe.instance.updatePlatformSheet(
      params: const PlatformPaySheetUpdateParams.applePay(
        summaryItems: [
          ApplePayCartSummaryItem.immediate(
            label: "Test-Item",
            amount: "19.90",
          )
        ],
        shippingMethods: [],
        errors: [],
      ),
    );
  },
  setOrderTracking: (String a, String b, String c, String d) {},
  onDidSetShippingContact: (contact) {
    Stripe.instance.updatePlatformSheet(
      params: const PlatformPaySheetUpdateParams.applePay(
        summaryItems: [
          ApplePayCartSummaryItem.immediate(
            label: "Test-Item",
            amount: "19.90",
          )
        ],
        errors: [],
        shippingMethods: [],
      ),
    );
  },
  type: PlatformButtonType.checkout,
  appearance: PlatformButtonStyle.whiteOutline,
  onPressed: () async {
    try {
      final token = "FILL_ME";
  
      // 2. Confirm apple pay payment
      await Stripe.instance.confirmPlatformPayPaymentIntent(
        clientSecret: token,
        confirmParams: const PlatformPayConfirmParams.applePay(
          applePay: ApplePayParams(
            shippingType: ApplePayShippingType.delivery,
            cartItems: [
              ApplePayCartSummaryItem.immediate(
                label: "Test-Item",
                amount: "19.90",
              )
            ],
            requiredShippingAddressFields: [],
            shippingMethods: [],
            merchantCountryCode: 'DE',
            currencyCode: 'EUR',
          ),
        ),
      );
  
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(
            content:
                Text('Apple Pay payment succesfully completed')),
      );
    } catch (e) {
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(content: Text('Error: $e')),
      );
      rethrow;
    }
  })
```

**Comment by remonh87** [comment]
@jonasbark  I fixed this one in the sync branch it looks like `onTrackShipping` returns a closure without arguments so when we force unwrap `args` it crashes.

**Comment by remonh87** [comment]
solved in version 9.1.1
